notebook/services/sessions/handlers.py

```diff
@@ -86,19 +86,46 @@ def get(self, session_id):
     @json_errors
     @gen.coroutine
     def patch(self, session_id):
-        # Currently, this handler is strictly for renaming notebooks
+        """Patch updates sessions:
+
+        - notebook.path updates session to track renamed notebooks
+        - kernel.name starts a new kernel with a given kernelspec
+        """
         sm = self.session_manager
+        km = self.kernel_manager
         model = self.get_json_body()
         if model is None:
             raise web.HTTPError(400, "No JSON data provided")
+
+        # get the previous session model
+        before = yield gen.maybe_future(sm.get_session(session_id=session_id))
+
         changes = {}
         if 'notebook' in model:
             notebook = model['notebook']
             if 'path' in notebook:
                 changes['path'] = notebook['path']
+        if 'kernel' in model:
+            if 'name' in model['kernel']:
+                kernel_name = model['kernel']['name']
+                kernel_id = yield sm.start_kernel_for_session(
+                    session_id, kernel_name=kernel_name, path=before['notebook']['path'])
+                changes['kernel_id'] = kernel_id
+            if 'id' in model['kernel']:
```

Comment:
I'm sending `null` if the `id` is not given, do you mind checking for that here?